packages/next/src/client/components/links.ts

```diff
@@ -31,18 +31,55 @@ type LinkInstance = {
   // The cache version at the time the task was initiated. This is used to
   // determine if the cache was invalidated since the task was initiated.
   cacheVersion: number
+
+  // Form also uses this instance but it does not have a setOptimisticLinkStatus.
+  setOptimisticLinkStatus: ((status: { pending: boolean }) => void) | null
+}
+
+// TODO: consider moving this to AppRouterState
+
+// Tracks the currently navigating link instance to manage loading states
+let linkForCurrentNavigation: LinkInstance | null = null
```

Comment:
I don't love the name of this because the navigation may have already finished a long time ago. It's more like `linkForMostRecentNavigation`.